src/test/java/org/alfresco/repo/security/person/PersonTest.java

```diff
@@ -1736,6 +1736,68 @@ public void testPreventCreationOfBuiltInAuthorities()
         {
             // expect to go here
         }
+    }
+
+    public void testUserShouldBeAbleToUpdateTheProfile()
+    {
+        final String USERNAME = GUID.generate();
+
+        PropertyMap guestProps = new PropertyMap();
+        guestProps.put(ContentModel.PROP_USERNAME, USERNAME);
+        guestProps.put(ContentModel.PROP_FIRSTNAME, "myFirstName");
+        guestProps.put(ContentModel.PROP_LASTNAME, "myLastName");
+        guestProps.put(ContentModel.PROP_EMAIL, "myFirstName.myLastName@email.com");
+
+        NodeRef personRef = personService.createPerson(guestProps);
+
+        AuthenticationUtil.setFullyAuthenticatedUser(USERNAME);
+
+        nodeService.setProperty(personRef, ContentModel.PROP_FIRSTNAME, "myUpdatedFirstName");
+    }
+
+    public void testPreventChangesToUsernameWithoutHavingAdminRights()
+    {
+        try
+        {
+            final String USERNAME = GUID.generate();
+            final String UPDATED_USERNAME = USERNAME + "1";
+
+            PropertyMap guestProps = new PropertyMap();
+            guestProps.put(ContentModel.PROP_USERNAME, USERNAME);
+            guestProps.put(ContentModel.PROP_FIRSTNAME, "myFirstName");
+            guestProps.put(ContentModel.PROP_LASTNAME, "myLastName");
+            guestProps.put(ContentModel.PROP_EMAIL, "myFirstName.myLastName@email.com");
 
+            NodeRef personRef = personService.createPerson(guestProps);

```

Comment:
this and most of the prev lines can be extracted into a method that can be used in the test above as well